src/core/Akka.NodeTestRunner/Program.cs

```diff
@@ -67,14 +85,60 @@ static void Main(string[] args)
                             specFail.FailureMessages.Add(ex.Message);
                             specFail.FailureStackTraces.Add(ex.StackTrace);
                             Console.WriteLine(specFail);
+
+                            //make sure message is send over the wire
+                            FlushLogMessages();
                             Environment.Exit(1); //signal failure
                         }
                         sink.Finished.WaitOne();
+
+                        FlushLogMessages();
+                        system.Shutdown();
+                        system.AwaitTermination();
+
                         Environment.Exit(sink.Passed ? 0 : 1);
                     }
                 }
             }
         }
+
+        private static void FlushLogMessages()
+        {
+            _logger.GracefulStop(TimeSpan.FromSeconds(2)).Wait();
```

Comment:
I implemented this blocking flush mechanism using `GracefulStop`, which uses a `PoisonPill` by default - this doesn't necessarily guarantee that the actor has pushed all of its messages onto the other end of the wire but it's a pretty good approximation.